### PR TITLE
Export components

### PR DESCRIPTION
**Pull request by lovincyrus**
closes https://github.com/allinbits/design/issues/170

### CODE REVIEW

**Comment by lovincyrus** [comment]
> I like the idea of re-exporting! However, "stars" are meant to import (or re-export) named exports, whereas we export `default`. Can't really re-export without having a name 🙂

Would re-exporting everything be easier to maintain down the road?

http://jamesknelson.com/re-exporting-es6-modules/
https://www.logicbig.com/tutorials/misc/javascript/transitive-export.html

**Comment by fadeev** [comment]
The approach I suggested is the same as the one in James K Nelson's article. I think it should work fine: one line per component.

I don't find repetition of component's name particularly appealing, however. We can rename `Modal/Modal.vue` to `Modal/index.vue` and use the following export:

```
export { default as Modal } from "./Modal";
```

If you're cool with having `index.vue` as a component name, we can use that.

**Comment by lovincyrus** [comment]
> PS I requested changes, because suggested in this PR approach does not work.

Interesting. It works for me locally and https://deploy-preview-130--cosmos-ui.netlify.com/. What errors are you seeing? 👀

**Comment by fadeev** [comment]
Deploy preview works, because that's Storybook. For Storybook an empty `src/index.js` works just as well, because it doesn't read this file. Storybook uses `.storybook/config.js`, which imports stories, which in turn import components. Try removing `src/index.js` locally and see that Storybook works perfectly fine.

Where proposed in this PR approach (`export * from ...`) doesn't work is when, say, `cosmos.network` tries to import components from `@cosmos-ui`. There error I'm seeing:

```
These relative modules were not found:
* ./Footer in ./node_modules/@cosmos-ui/vue/src/index.js...
```

In short:

```
export * from "./Modal"; // Doesn't work -- can't use this

export { default as Modal } from "./Modal/Modal"; // Works fine
export { default as Modal } from "./Modal"; // Works, if you rename Modal.vue to index.vue
```
